### ISSUE TITLE
Handling of char *

### ISSUE THREAD

**Issue by obonaventure** [pgbovine/opt-cpp-backend#70]
There is an issue in how python tutor visualises the following C code

```c 
int main() {
  char *s1="ABCDEFGH";
  char s2[9]="ABCDEFGH";
  s2[0]='$';
  s1[0]='$';
  return 0;
}
```

s1 is shown to be a pointer some memory on the heap while in fact C compilers will store the "ABCDEF" string in the dss. Since this DSS is usually a read-only region of memory, the line s1[0]='$'; causes a segfault which cannot be understood from the visualisation

**Comment by pgbovine** [comment]
good point; there was an earlier bug report about this too, so I will make a note of it. I should have a separate display for DSS, separate from heap
